datasets/noaa/pipelines/_images/run_csv_transform_kub/csv_transform.py

```diff
@@ -0,0 +1,1043 @@
+    start_year: str,
+    input_csv_headers: typing.List[str],
+    data_dtypes: dict,
+    reorder_headers_list: typing.List[str],
+    null_rows_list: typing.List[str],
+    date_format_list: typing.List[str],
+    slice_column_list: dict,
+    regex_list: dict,
+    rename_headers_list: dict,
+    remove_source_file: str,
+    delete_target_file: str,
+    number_of_header_rows: str,
+    int_date_list: typing.List[str],
+    gen_location_list: dict,
+) -> None:
+    logging.info(f"{pipeline_name} process started")
+    pathlib.Path("./files").mkdir(parents=True, exist_ok=True)
+    execute_pipeline(
+        pipeline_name=pipeline_name,
+        source_url=source_url,
+        source_file=source_file,
+        target_file=target_file,
+        chunksize=chunksize,
+        ftp_host=ftp_host,
+        ftp_dir=ftp_dir,
+        project_id=project_id,
+        dataset_id=dataset_id,
+        destination_table=table_id,
+        target_gcs_bucket=target_gcs_bucket,
+        target_gcs_path=target_gcs_path,
+        schema_path=schema_path,
+        drop_dest_table=drop_dest_table,
+        input_field_delimiter=input_field_delimiter,
+        ftp_batch_size=ftp_batch_size,
+        ftp_batch_sleep_time=ftp_batch_sleep_time,
+        full_data_load=full_data_load,
+        start_year=start_year,
+        input_csv_headers=input_csv_headers,
+        data_dtypes=data_dtypes,
+        reorder_headers_list=reorder_headers_list,
+        null_rows_list=null_rows_list,
+        date_format_list=date_format_list,
+        slice_column_list=slice_column_list,
+        regex_list=regex_list,
+        rename_headers_list=rename_headers_list,
+        remove_source_file=(remove_source_file == "Y"),
+        delete_target_file=(delete_target_file == "Y"),
+        number_of_header_rows=int(number_of_header_rows),
+        int_date_list=int_date_list,
+        gen_location_list=gen_location_list,
+    )
+    logging.info(f"{pipeline_name} process completed")
+
+
+def execute_pipeline(
+    pipeline_name: str,
+    source_url: str,
+    source_file: pathlib.Path,
+    target_file: pathlib.Path,
+    chunksize: str,
+    ftp_host: str,
+    ftp_dir: str,
+    project_id: str,
+    dataset_id: str,
+    destination_table: str,
+    target_gcs_bucket: str,
+    target_gcs_path: str,
+    schema_path: str,
+    drop_dest_table: str,
+    input_field_delimiter: str,
+    ftp_batch_size: str,
+    ftp_batch_sleep_time: str,
+    full_data_load: str,
+    start_year: str,
+    input_csv_headers: typing.List[str],
+    data_dtypes: dict,
+    reorder_headers_list: typing.List[str],
+    null_rows_list: typing.List[str],
+    date_format_list: typing.List[str],
+    slice_column_list: dict,
+    regex_list: dict,
+    remove_source_file: bool,
+    rename_headers_list: dict,
+    delete_target_file: bool,
+    number_of_header_rows: int,
+    int_date_list: typing.List[str],
+    gen_location_list: dict,
+) -> None:
+    if pipeline_name == "GHCND by year":
```

Comment:
Perhaps "GHCND by year" can be defined as a constant at the top of this file (or better yet, read from a yaml file). Same goes for other similar cases.

Reply:
What added value would that bring to the code?  What would the variable be called?